### DISCUSSION TITLE
Error ketika deploy ke Shiny

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#508] [Q&A] [answered]
<img width="376" alt="image" src="https://user-images.githubusercontent.com/106010284/211732618-5bf2040f-67d7-4b37-ba8b-17d7d7e79c49.png">

Saya mengalami error ketika ingin deploy project ke shiny. Mohon bantuannya kak.

**Comment by ranydwi** [comment]
Dear @yohanespm77  ,

Untuk mengetahui informasi error yang terjadi, silahkan untuk melakukan pengecekan pada log shinyapps yang dideploy pada akun shinyapps.io Anda. Boleh dicek pada bagian log app terbawah jika terdapat pesan error, maka Anda perlu memperbaiki error yang terjadi tersebut.

Selain itu izin memastikan apakah saat menjalankan dashboard secara lokal mengalami error yang sama? atau hal ini hanya terjadi saat melakukan deployment di shinyapps?

**Reply by yohanespm77** [reply]
<img width="708" alt="image" src="https://user-images.githubusercontent.com/106010284/211772533-db43137a-2a5d-4d76-adb0-bc147c131ef9.png">

Ga ada pesan errornya kak. 
Saat menjalankan dashboard di lokal, tidak ada error. Dashboardnya bisa tampil secara utuh.

**Reply by ranydwi** [reply]
Dear @yohanespm77 

Kalau dari log yang tertampil di baris kedua terakhir tertampil `out of memory`. Hal ini sepertinya dikarenakan out of memory pada RAM anda. Saya pikir Anda mungkin memuat terlalu banyak package, mungkin lebih dari yang sebenarnya dibutuhkan oleh kode Anda.

Berdasarkan beberapa referensi, Anda dapat mencoba menambah ukuran memori dengan menggunakan kode berikut: `rsconnect::configureApp("APPNAME", size="xlarge")`

Referensi: https://stackoverflow.com/questions/63145590/shiny-io-out-of-memory 

Semoga membantu.

**Reply by yohanespm77** [reply]
<img width="329" alt="image" src="https://user-images.githubusercontent.com/106010284/211778386-2f3f04f3-aa06-47cf-9af0-93b339e0411b.png">

Packagenya hanya itu kak. 


Yg bagian rsconnect, errornya spt ini kak :
<img width="388" alt="image" src="https://user-images.githubusercontent.com/106010284/211778713-68d7041e-5f2c-4c89-8284-b7562e15ea05.png">

**Reply by ranydwi** [reply]
Dear @yohanespm77 ,

Untuk "APPNAME" pada configureApp silahkan sesuaikan dengan nama app pada Shiny Bapak.
Referensi yang bisa bapak baca: https://shiny.rstudio.com/articles/shinyapps.html

Terima kasih

**Reply by yohanespm77** [reply]
<img width="628" alt="image" src="https://user-images.githubusercontent.com/106010284/211975338-490c7179-372d-41f1-b0c2-e34d63b05c09.png">

Not allowed kak. 

Saya coba size = large, tetep sama errorya, signal:killed.

**Comment by ranydwi** [comment]
Dear @yohanespm77 

Baik pak kalau memang error menggunakan rsconnect, silahkan set size shiny.maxRequestSize menggunakan code berikut `options(shiny.maxRequestSize=200*1024^2)` pada bagian atas file ui.R maupun server.R Bapak.

Semoga membantu.
Team Algoritma

**Reply by yohanespm77** [reply]
<img width="293" alt="image" src="https://user-images.githubusercontent.com/106010284/211982247-d8623445-b788-4946-b88c-0853b0e976b3.png">

Udah saya tambahkan codenya kak, tapi masih signal:killed
<img width="383" alt="image" src="https://user-images.githubusercontent.com/106010284/211982355-2dc18ab0-3ce9-4f7a-9bd7-3ff40c90cf64.png">

Datanya udh saya ubah jg ke format rds, ukurannya jadi 12.5mb

**Comment by hwulanayu** [comment]
Dear @yohanespm77 

Error yang terjadi di shiny Bapak hanya ketika deployment, bukan di local. Bapak dapat mencoba beberapa cara:
- cek apakah ada objek dengan ID yang sama: [referensi stackoverflow](https://stackoverflow.com/questions/68385614/shiny-the-application-failed-to-start-exited-normally-with-code-137-signal).
- cek apakah ada karakter khusus pada dataset yang belum dihandle dengan encoding: [referensi stackoverflow](https://stackoverflow.com/questions/68385614/shiny-the-application-failed-to-start-exited-normally-with-code-137-signal).
- cek konsistensi tipe data dan global environmen: [referensi rstudio community](https://community.rstudio.com/t/shinyapps-io-returning-an-error-upon-deployment/142515/10).
- jika kondisi-kondisi di atas sudah tidak ada masalah dan deployment masih mengeluarkan error, coba buat project baru dengan folder yang baru. Salin semua kode ke dalam folder dan file project baru dan deploy kembali project Bapak.  

Semoga membantu!
Team Algoritma

**Reply by yohanespm77** [reply]
Terima kasih atas masukannya kak. Saya sudah coba buat project baru dgn folder baru, akan tetapi kendalanya tetap sama (out of memory). 

Saya coba memotong data saya sehingga ukurannya menjadi 5.64mb. Saya coba deploy ulang, berhasil.
Akan tetapi saya butuh mendeploy project dengan data awal yg ukurannya 12.5mb namun selalu gagal (out of memory).

Apakah ada solusinya ?

**Reply by ranydwi** [reply]
Dear Pak @yohanespm77 

Baik pak. Saya menemukan discussion yang sama persis masalahnya dengan bapak yang berkaitan dengan dataset yang cukup besar, ternyata Pak untuk yang mencoba mengetikan Rsconnect kemarin silahkan untuk mengetikkannya di **console.**

`rsconnect::configureApp("Cyber_Threats", size="xxlarge")`

Semoga kali ini permasalahannya solved ya Pak.
Source: https://community.rstudio.com/t/size-of-shiny-application-for-deployment-out-of-memory/144342

Regards.

**Reply by yohanespm77** [reply]
Dear Kak @ranydwi 

Saya coba mengetikkan codenya di console, tapi masih error kak (not allowed).

<img width="660" alt="image" src="https://user-images.githubusercontent.com/106010284/212597728-e1c35293-e9b8-436c-97dd-f0906ce9793c.png">

**Comment by ramdhinafinita** [comment]
Dear Pak @yohanespm77

Sebelumnya, Saya ingin bertanya terlebih dahulu. Berdasarkan error log pada gambar berikut, terdapat error pada container pertama di shinny Bapak. Pada container tersebut, konten apa yang Bapak masukkan? apakah model (model machine learning) ataukah visualisasi (seperti ggplot, plotly)?

![image](https://user-images.githubusercontent.com/64061969/212613542-dc6fa829-2272-44a2-983f-571898a25bb7.png)

Regards.

**Reply by yohanespm77** [reply]
Dear Kak @ramdhinafinita 

Pada gambar tersebut, content yg saya masukkan visualisasi plotly kak.

**Comment by ramdhinafinita** [comment]
Dear Pak @yohanespm77 

Berdasarkan error Bapak, berarti memang dikarenakan memory pada container tersebut melebihi yang batas maksimal yakni 5 mb. Salah satu solusi yang bisa kami berikan yakni bisa dengan mengubah data awal bapak menjadi format .RDS 

Terimakasih

**Reply by yohanespm77** [reply]
Dear Kak @ramdhinafinita 

Datanya sudah saya ubah ke format .RDS kak, ukurannya jadi 12.5mb, ga bisa di bawah 5mb. Apakah ada solusi lain terkait hal ini ?

**Reply by ramdhinafinita** [reply]
Dear Pak @yohanespm77

Terima kasih atas pertanyaan yang disampaikan, pertanyaan Anda sudah kami terima. Kami akan mencoba membantu mencarikan referensi berkaitan dengan pertanyaan yang diberikan.

**Comment by VictorNugraha** [comment, marked as answer]
Dear Pak @yohanespm77,

Salah satu opsi yang dapat kami berikan ketika mengalami limitasi data dengan size yang cukup besar adalah dengan meng-upload datanya ke googlesheet dan aplikasi shiny akan membaca datanya dari googlesheet tersebut.

Berikut kami lampirkan referensi yang dapat dicoba dalam melakukan proses pembacaan data melalui googlesheet oleh shiny, https://debruine.github.io/shinyintro/data.html 

Demikian informasi yang dapat kami sampaikan, selamat mencoba dan semoga berhasil.

Best Regards,

Team Algoritma

**Reply by ramdhinafinita** [reply]
Halo Pak @yohanespm77 , untuk menjaga agar forum diskusi berjalan kondusif, karena tidak ada respon lanjutan setelah 7 hari dari respon terakhir, kami izin menandai pertanyaan ini sudah terjawab dan menutup diskusi ini. Semoga dapat dipahami dan terima kasih.

**Reply by yohanespm77** [reply]
Baik, terima kasih kak.
